main.cpp

```diff
@@ -13,75 +13,584 @@ What you create in this project will be used as the basis of Project 5 in the co
+
+        //3) a member function whose parameter has a default value.
+        //the parameter name is related to the work the function will perform.
+        void fillTank(double fuelAmountInGallons = 2.0);  
+        void breakDown(std::string failureType, bool requiresTow = false);
+        int getMilesTraveledAnnually(bool includeUberLyftTrips);
+    };
+
+    //wash and wax car
+    void washAndWaxCar( Car car ); //4) a member function whose parameter is a UDT.
+    //charge customer
+    float chargeCustomer(float discountPercentage);
+    //detail the car interior
+    void detailInterior( Car car );
+    
+    //5) a member variable whose type is a UDT.
+    Car carBeingServiced;  
+};
+
+
+/*
+Thing 1) Button Factory
+5 properties:
+    1) Number of workers
+    2) Number of buttons per month manufactured
+    3) Number of button types   
+    4) Average cost of button
+    5) Profit per quarter
+3 things it can do:
+    1) Manufacture buttons
+    2) Design buttons
+    3) Ship buttons to businesses
+ */
+struct ButtonFactory
+{
+ //   1) Number of workers
+    int numOfWorkers = 30;
+ //   2) Number of buttons per month manufactured
+    int numOfButtonsPerMo = 10500;
+ //   3) Number of button types
+    int numOfButtonTypes = 107;   
+ //   4) Average cost of button
+    float avgCostPerButton = 0.03f;
+ //   5) Profit per quarter
+    float profitPerQtr = 9450.00f;
+//3 things it can do:
+ //   1) Manufacture buttons
+    void makeButton(int buttonType);
+ //   2) Design buttons
+    void designButton();
+ //   3) Ship buttons to businesses
+    void shipButtons(std::string address, int buttonType, int numToShip);
+};
+/*
+Thing 2) Alien detection agency
+5 properties:
+    1) number of aliens on ground
+    2) fleet size
+    3) intelligence (IQ)
+    4) global coordinates detected from
+    5) case number
+3 things it can do:
+    1) process reports by field agents
+    2) filter out false reports
+    3) publish vetted reports
+ */
+
+struct AlienDetectionAgency //this is slightly revised from 1d
+{
+    //1) number of aliens on ground
+    int aliensOnGround = 731;
+    //2) fleet size
+    int alienFleetSize = 437;
+    //3) intelligence (IQ)
+    int avgAlienIntelligence = 130;
+    //4) Total Field Reports
+    unsigned int totalFieldReports = 1247;
+    //5) Total Field Agents
+    unsigned int totalFieldAgents = 12541;
+
+    struct FieldReport
+    {
+        //5) case number
+        unsigned int caseNumber = 1296;
+        //4) global coordinates detected from
+        struct Coordinate
+        {
+            double latitude = 0.0;
+            double longitude = 0.0;
+        };
+        Coordinate detectionCoord;
+        
+        std::string agentFirstName;
+        std::string agentLastName;
+        std::string eventDescription;
+        int numOfWitnesses;
+        bool verified = false;
+
+    };
+    //1) process reports by field agents
+    void ReportHandler(FieldReport currReport)
```

Comment:
member functions follow the same naming rules as member variables:
- non-type names start with a lower case letter and use `camelCase`
- Type names start with an upper-case letter and use `camelCase`
- [ ] revise